compiler/src/dotty/tools/dotc/transform/OverridingPairs.scala

```diff
@@ -20,14 +20,15 @@ object OverridingPairs {
   /** The cursor class
    *  @param base   the base class that contains the overriding pairs
    */
-  class Cursor(base: Symbol)(implicit ctx: Context) {
+  class Cursor(base: Symbol, excludeArtifacts: Boolean = true)(implicit ctx: Context) {
```

Comment:
This seems a bit roundabout to me. Instead of creating symbols for artifacts just to ignore that afterwards, I think we should just not create symbols for artifacts in ClassfileParser. Given that we don't see these symbols under joint compilation it should reduce the differences in execution between joint and separate compilation and thus reduce the number of unique bugs to one or the other.

Reply:
I propose to make this into a separate issue since it seems to be a tricky topic.